lib/contracts/message-reaction.ts

```diff
@@ -0,0 +1,36 @@
+import type { ContractDefinition } from '@balena/jellyfish-types/build/core';
+import { contractMixins } from 'autumndb';
+
+export const messageReaction: ContractDefinition = {
+	slug: 'message-reaction',
+	type: 'type@1.0.0',
+	name: 'message reaction',
+	markers: [],
+	data: {
+		schema: {
+			type: 'object',
+			required: ['version', 'data'],
+			properties: {
+				version: {
+					type: 'string',
+					const: '1.0.0',
+				},
+				data: {
+					type: 'object',
+					properties: {
+						actor: {
+							type: 'string',
+							format: 'uuid',
+						},
+						reaction: {
+							type: 'string',
+							format: 'unicode',
+						},
+					},
+					required: ['actor', 'reaction'],
+				},
+			},
+		},
+		indexed_fields: [['data.payload.alertsUser'], ['data.actor']],
```

Comment:
I'm guessing you're still in the middle of working on this, but I'll comment before I forget. You probably don't need either of these. Especially the first as its referencing a field in the contract that doesn't even exist. This `indexed_fields` bit triggers the backend to create partial indexes using the paths defined as its value.